modules/layers/src/solid-polygon-layer/polygon-tesselator.js

```diff
@@ -24,62 +24,30 @@
 // - 3D surfaces (top and sides only)
 // - 3D wireframes (not yet)
 import * as Polygon from './polygon';
-import {experimental} from '@deck.gl/core';
-const {fillArray} = experimental;
+import Tesselator from '../path-layer/tesselator';
 import {fp64 as fp64Module} from 'luma.gl';
 const {fp64LowPart} = fp64Module;
 
-// Maybe deck.gl or luma.gl needs to export this
-function getPickingColor(index) {
-  index++;
-  return [index & 255, (index >> 8) & 255, (index >> 16) & 255];
-}
-
-const DEFAULT_COLOR = [0, 0, 0, 255]; // Black
-
 // This class is set up to allow querying one attribute at a time
 // the way the AttributeManager expects it
-export class PolygonTesselator {
-  constructor({polygons, IndexType}) {
-    // Normalize all polygons
-    polygons = polygons.map(polygon => Polygon.normalize(polygon));
-
-    // Count all polygon vertices
-    const pointCount = getPointCount(polygons);
-
-    this.polygons = polygons;
-    this.pointCount = pointCount;
-    this.bufferLayout = polygons.map(polygon => Polygon.getVertexCount(polygon));
-    this.IndexType = IndexType;
-
-    // TODO: dynamically decide IndexType in tesselator?
-    // Check if the vertex count excedes index type limit
-    if (IndexType === Uint16Array && pointCount > 65535) {
-      throw new Error("Vertex count exceeds browser's limit");
-    }
-
-    this.attributes = {
-      pickingColors: calculatePickingColors({polygons, pointCount})
-    };
-  }
-
-  updatePositions({fp64, extruded}) {
-    const {attributes, polygons, pointCount} = this;
-
-    attributes.positions = attributes.positions || new Float32Array(pointCount * 3);
-    attributes.vertexValid = attributes.vertexValid || new Uint8ClampedArray(pointCount).fill(1);
-
-    if (fp64) {
-      // We only need x, y component
-      attributes.positions64xyLow = attributes.positions64xyLow || new Float32Array(pointCount * 2);
-    }
-
-    updatePositions({cache: attributes, polygons, extruded, fp64});
+export default class PolygonTesselator extends Tesselator {
+  constructor({data, getGeometry, fp64, IndexType = Uint32Array}) {
+    super({
+      data,
+      getGeometry,
+      fp64,
+      attributes: {
+        positions: {size: 3},
+        positions64xyLow: {size: 2},
+        vertexValid: {type: Uint8ClampedArray, size: 1},
+        indices: {type: IndexType, size: 1}
```

Comment:
For longer term: 

* Perhaps we could unburden the layers (and/or their custom tesselators) from dealing with IndexType. We could just specify `indices: true` here and let the `Tesselator` base class select type as appropriate based on number of vertices.